src/main/java/com/conveyal/osmlib/VanillaExtract.java

```diff
@@ -71,16 +66,14 @@ public static void main(String[] args) {
         try {
             httpServer.start();
             LOG.info("VEX server running.");
-            Thread.currentThread().join();
-//            updateThread.interrupt();
+            //Thread.currentThread().join();
```

Comment:
I did this because it was causing the VanillaExtractTest to hang when it tried to start the server.  The test would not continue after the server started up.

Reply:
I don't think we should be changing the flow of our main method in order to make a test easier to write. We need to establish clearly what we're losing by not doing the join and clean shutdown -- this doesn't seem like the kind of thing that should just be changed without understanding the consequences.

Reply:
I'm not a java expert so I don't understand what the thread joining does.

Reply:
`Thread.currentThread().join();` _is_ kind of strange. It just blocks the current thread forever, waiting for itself to end. It only makes sense because this is intended to be run as a command line main method. Once the server thread starts, we don't want the main thread to shut down and end the whole process. This is another reason that if we want to test this logic, it should be factored out and called slightly differently from a test and from a main method.